fastlane/lib/fastlane/actions/slack.rb

```diff
@@ -1,70 +1,178 @@
-        message = message[start_index..-1]
-        # We want line breaks to be shown on slack output so we replace
-        # input non-interpreted line break with interpreted line break
-        message.gsub('\n', "\n")
-      end
+        def run(options)
+          options[:message] = self.trim_message(options[:message].to_s || '')
+          options[:message] = Fastlane::Notification::Slack::LinkConverter.convert(options[:message])
 
-      def self.run(options)
-        require 'slack-notifier'
+          options[:pretext] = options[:pretext].gsub('\n', "\n") unless options[:pretext].nil?
+
+          if options[:channel].to_s.length > 0
+            channel = options[:channel]
+            channel = ('#' + options[:channel]) unless ['#', '@'].include?(channel[0]) # send message to channel by default
+          end
 
-        options[:message] = self.trim_message(options[:message].to_s || '')
-        options[:message] = Slack::Notifier::Util::LinkFormatter.format(options[:message])
+          username = options[:use_webhook_configured_username_and_icon] ? nil : options[:username]
 
-        options[:pretext] = options[:pretext].gsub('\n', "\n") unless options[:pretext].nil?
+          slack_attachment = generate_slack_attachments(options)
+          link_names = options[:link_names]
+          icon_url = options[:use_webhook_configured_username_and_icon] ? nil : options[:icon_url]
 
-        if options[:channel].to_s.length > 0
-          channel = options[:channel]
-          channel = ('#' + options[:channel]) unless ['#', '@'].include?(channel[0]) # send message to channel by default
+          post_message(
+            channel: channel,
+            username: username,
+            attachments: [slack_attachment],
+            link_names: link_names,
+            icon_url: icon_url,
+            fail_on_error: options[:fail_on_error]
+          )
         end
 
-        username = options[:use_webhook_configured_username_and_icon] ? nil : options[:username]
+        def post_message(channel:, username:, attachments:, link_names:, icon_url:, fail_on_error:)
+          @notifier.post_to_legacy_incoming_webhook(
+            channel: channel,
+            username: username,
+            link_names: link_names,
+            icon_url: icon_url,
+            attachments: attachments
+          )
+          UI.success('Successfully sent Slack notification')
+        rescue => error
+          UI.error("Exception: #{error}")
+          message = "Error pushing Slack message, maybe the integration has no permission to post on this channel? Try removing the channel parameter in your Fastfile, this is usually caused by a misspelled or changed group/channel name or an expired SLACK_URL"
+          if fail_on_error
+            UI.user_error!(message)
+          else
+            UI.error(message)
+          end
+        end
 
-        notifier = Slack::Notifier.new(options[:slack_url], channel: channel, username: username)
```

Comment:
we could have kept the old implementation (all helper methods) and just removed `require 'slack-notifier'` with this one line to below Fastlane to minimize the breaking changes
`notifier = Fastlane::Notification::Slack.new(options[:slack_url])`